engine/src/main/java/org/terasology/engine/rendering/nui/layers/mainMenu/WorldSetupScreen.java

```diff
@@ -54,82 +51,42 @@ public class WorldSetupScreen extends CoreScreenLayer {
     private WorldConfigurator oldWorldConfig;
-    private Name newWorldName;
 
     @Override
     public void initialise() {
         setAnimationSystem(MenuAnimationSystems.createDefaultSwipeAnimation());
 
-        WidgetUtil.trySubscribe(this, "close", button -> {            
-            UIText customWorldName = find("customisedWorldName", UIText.class);
-
-            boolean goBack = false;
-
-            //sanity checks on world name
-            if (customWorldName.getText().isEmpty()) {
-                //name empty: display a popup, stay on the same screen
-                getManager().pushScreen(MessagePopup.ASSET_URI, MessagePopup.class)
-                        .setMessage("Name Cannot Be Empty!", "Please add a name for the world");
-            } else if (customWorldName.getText().equalsIgnoreCase(world.getWorldName().toString())) {
-                //same name as before: go back to universe setup
-                goBack = true;
-            } else {
-                //no match found: go back to universe setup
-                goBack = true;
-            }
+        WidgetUtil.trySubscribe(this, "close", button -> {
 
-            if (goBack) {
-                newWorldName = new Name(customWorldName.getText());
-                world.setWorldName(newWorldName);
-                triggerBackAnimation();
-            }
+            triggerBackAnimation();
         });
     }
-
-    /**
-     * This method sets the world name in title as well as in UITextBox
-     *
-     * @param customWorldName
-     */
-    private void setCustomWorldName(UIText customWorldName) {
-        customWorldName.setText(world.getWorldName().toString());
-    }
-
     @Override
     public void onOpened() {
         super.onOpened();
 
-        UILabel subitle = find("subtitle", UILabel.class);
-        subitle.setText(translationSystem.translate("${engine:menu#world-setup}") + " for " + world.getWorldName().toString());
-        UIText customWorldName = find("customisedWorldName", UIText.class);
-        setCustomWorldName(customWorldName);
+        UILabel subtitle = find("subtitle", UILabel.class);
+        subtitle.setText(translationSystem.translate("${engine:menu#world-setup}") + " for " + universe.getGameName().toString());
     }
 
     /**
      * This method sets the world whose properties are to be changed. This function is called before the screen comes
      * to the forefront.
      *
      * @param subContext    the new environment created in {@link UniverseSetupScreen}
-     * @param worldSelected the world whose configurations are to be changed.
+     * @param universe      the universe whose world's configurations are to be changed.
      * @throws UnresolvedWorldGeneratorException
      */
-    public void setWorld(Context subContext, WorldSetupWrapper worldSelected)
-            throws UnresolvedWorldGeneratorException {
-        world = worldSelected;
+    public void setWorld(Context subContext, UniverseWrapper universe) {
```

Comment:
I'm annoyed by these HiddenFieldChecks - I'd like to have an overview in how many places this would actually cause trouble right now, and if we find none, disable this check.